[Julio Harjo] iP

junlee1991 opened the pull request:
### Duke Application
"Your mind is for having ideas, not holding them." - David Allen


"Your mind is for having ideas, not holding them." - David Allen
The Duke application helps frees your mind of having to remember things you need to do. It's,

The Duke application helps frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPERFAST to use!

All you need to do is,

1.download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html)
2.double click it
3.add your tasks
4.let it manage your tasks for you 👍

And it is **FREE**!

Features:

 Managing Tasks
 Managing Events
 Reminders! (Coming soon)
If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

junlee1991 committed:
Level-1 done

junlee1991 committed:
Level-2 done

junlee1991 committed:
Level-3 done

junlee1991 committed:
Level-4 done

junlee1991 committed:
A-TextUiTesting done

junlee1991 committed:
Level-5 done

junlee1991 committed:
Level-6 done

junlee1991 committed:
Level-7 done

junlee1991 committed:
Level-8 done

junlee1991 committed:
Merge branch 'branch-Level-8'

junlee1991 committed:
Level-8 done

junlee1991 committed:
Revert "Level-8 done"

This reverts commit 74bb18516436430981670d6bd91033f647749f83.

junlee1991 committed:
A-MoreOOP done

junlee1991 committed:
A-Packages done

junlee1991 committed:
Delete TaskManager

junlee1991 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

junlee1991 committed:
A-JUnit done

junlee1991 committed:
A-Jar done

junlee1991 committed:
A-JavaDoc done

junlee1991 committed:
A-CodingStandard done

junlee1991 committed:
Level-9 done

junlee1991 committed:
Fix import statements

junlee1991 committed:
Fix javaDocs

junlee1991 committed:
Fix coding standard

junlee1991 committed:
Fix level-9

junlee1991 committed:
Revert "Fix javaDocs"

This reverts commit 1c499180c6bfa080b8beebd4584e6831996b8153.

junlee1991 committed:
Fix javaDoc

junlee1991 committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

junlee1991 committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

junlee1991 committed:
Merge remote-tracking branch 'origin/branch-Level-9'

Mr-Teal commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+                    com = new ListCommand();
+                    break;
+                case ("find"):
+                    checkIfBlank(c);
```
My suggestion is that all the validity checking work can be put inside the corresponding command classes, but no big deal.

Mr-Teal commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+                    throw new InvalidInstructionException();
+            }
+        } catch (NumberFormatException e) {
+            throw new NumberFormatException("some msg");
```
I am a bit confused by the code here. Is this part necessary? If you think so then never mind.

Mr-Teal commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,140 @@
+/**
+ * Storage class to handle saving and writing data
+ */
+public class Storage {
```
Complete and perfect saving and loading task list in this class. I think it is good!

Mr-Teal reviewed (commented):
Overall the entire program seems nice.

ryanjansen commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        input = input.substring(9);
```
Could you put this logic in the parser instead? The abstraction seems to make more sense there.

ryanjansen commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws DukeException{
+        ArrayList<Task> res = new ArrayList<>();
```
Could this be shifted to the TaskList class?

ryanjansen reviewed (approved):
Code looks good, just that the abstraction can be more cleanly maintained by abstracting logic out of the command classes into the respective classes like the Parser and Tasklist

junlee1991 committed:
Level-10 done

junlee1991 committed:
Level-10 done

junlee1991 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

junlee1991 committed:
A-Checkstyle partially done

junlee1991 committed:
A-Checkstyle partially done

junlee1991 committed:
A-Checkstyle done

junlee1991 committed:
Update build.gradle

.jar file generated uses CLI as it uses duke.Duke as the main class.

Let's change the build.gradle application main class to duke.Launcher
to use the GUI instead.

junlee1991 committed:
Update Parser.java for A-Assertions

Parser parses the input by commands

Let's add assertions to ensure that the inputs are parsed correctly
according to the number of substrings in that command

junlee1991 committed:
Delete dead code and fix checkstyle

junlee1991 committed:
Add some assertions

junlee1991 committed:
Merge pull request #2 from junlee1991/branch-A-Assertions

Update Storage.java for A-Assertions

junlee1991 committed:
Merge pull request #1 from junlee1991/branch-A-CodeQuality

Updated files for A-CodeQuality

junlee1991 committed:
C-DetectDuplicates done

dohaduong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,144 @@
+            case ("mark"):
+                checkIfBlank(c);
+                checkIfValidInteger(c);
+                assert !command.substring(5).isEmpty();
```
Maybe you can change 5 to a static constant (magic number)

dohaduong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,144 @@
+            case ("delete"):
+                checkIfBlank(c);
+                checkIfValidInteger(c);
+                assert !command.substring(7).isEmpty();
```
Magic number (7)

junlee1991 committed:
Write user guide

junlee1991 committed:
Update readme.md
